### PR TITLE
[rtl/spi_device] Add extra cycle for CDC in spid_addr_4b

### PR DESCRIPTION
**Pull request by matutem**
The update mechanism uses a stable value after the spi_clk synchronizers.

### CODE REVIEW

File: hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson [hunk 1]
```diff
@@ -72,7 +72,7 @@ name:
 
   // TODO(lowrisc/opentitan#16689): Enable cdc instrumentation
   run_opts: ["+otbn_elf_dir={otbn_elf_dir}",
-             "+cdc_instrumentation_enabled=0"]
+             "+cdc_instrumentation_enabled=1"]
```
Comment (a-will) [inline hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson:75]: nit: This change should be left out or put into a different commit ;)

**Review by a-will — APPROVED** [review]
So the problem is that these two events race each other:
```
# 2 or 3 cycles
reg2hw_cfg_addr_4b_en_q_i -> spi_reg_cfg_addr_4b_en_sync

# 2 or 3 cycles
sck_csb (from spi_device.u_csb_edge_spiclk) -> spi_csb_asserted_pulse_i
```

And so, this mechanism fails when `spi_csb_asserted_pulse_i` takes 2 cycles and `spi_reg_cfg_addr_4b_en_sync` takes 3 cycles. Basically, there was a poor choice of sampling point in the code prior to this change, and this ECO will fix that up so at least the timing works out?

LGTM. Thanks for the fix!

File: hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson (hunk 1 above)
Comment (msfschaffner) [reply hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson:75]: @matutem can you remove this change in this PR?

**Comment by msfschaffner** [comment]
Pending feasibility analysis.
@arnonsha can you take a look and assess?

**Review by msfschaffner — APPROVED** [review]

File: hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson (hunk 1 above)
Comment (matutem) [reply hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson]: Enabling otbn cdc was a slip. Removed from this PR.

**Comment by matutem** [comment]
We need an issue describing the bug, and a clean spi_device block level regression with cdc instrumentation enabled.
@hcallahan-lowrisc, could you file the issue and run a regression with this fix?

File: hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson (hunk 1 above)
Comment (msfschaffner) [reply hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson]: Thanks!

**Comment by msfschaffner** [comment]
The issue has been filed here already https://github.com/lowRISC/opentitan/issues/19005

**Comment by matutem** [comment]
At full chip this fixes chip_sw_spi_device_pass_through when run with cdc instrumentation enabled in chip_sim_cfg.hjson
  run_opts: ["+cdc_instrumentation_enabled=1"]

### Test Results
|  Stage  |              Name               | Tests                           |  Max Job Runtime  |  Simulated Time  |  Passing  |  Total  |  Pass Rate  |
|:-------:|:-------------------------------:|:--------------------------------|:-----------------:|:----------------:|:---------:|:-------:|:-----------:|
|   V2    | chip_sw_spi_device_pass_through | chip_sw_spi_device_pass_through |      3.008m       |     9.101ms      |     8     |    8    |  100.00 %   |

So we are just missing a clean block level regression.

**Comment by matutem** [comment]
The spi_device block level full regression failed last night. It may be best to run it in a more hefty environment, like Google's.

**Comment by a-will** [comment]
Hm... Now that I'm thinking about this, the CDC infrastructure is actually testing an invalid timing mode, and the ECO may be unnecessary.

You see, software will only ever change the address mode when there is no chance of a SPI clock arriving anytime soon (i.e. it would be a violation of supported use cases to change the address mode underneath an active host). So it will never "just miss" the first clock through the synchronizer, and software-initiated updates should _always_ take 2 cycles through the flops. Assuming I've done my calculations correctly, that is, hehe.

**Review by crteja — APPROVED** [review]

**Review by hcallahan-lowrisc — APPROVED** [review]

**Comment by msfschaffner** [comment]
CHANGE AUTHORIZED: hw/ip/spi_device/rtl/spid_addr_4b.sv

**Comment by moidx** [comment]
CHANGE AUTHORIZED: hw/ip/spi_device/rtl/spid_addr_4b.sv

**Review by moidx — APPROVED** [review]

**Comment by msfschaffner** [comment]
We will merge this now so that DV and signoff can make progress.
@arnonsha if there is any issue with this ECO, let us know and we'll revert.
